comms/DiskAttribute.cpp

```diff
@@ -336,6 +337,19 @@ class CDiskAttribute : public CAttributeBase, public IDiskAttribute, public IAtt
         return newAttr;
     }
 
+    IAttribute * ChangeAttributeType(const TCHAR* label, const TCHAR* ext)
+    {
+        if (IsValidExtension(ext)) {
+            DeleteAttribute(this);
+            m_type = CreateIAttributeType(ext);
+        }
+        else {
+            return NULL;
+        }
+
+        return Rename(label);
```

Comment:
Your using "Rename" to trigger the notification cycle which is ok, but I would add a comment to Rename or a second param to accept a new file ext to explain why a Rename is being called with the same label as the current one.